irohad/ordering/impl/ordering_service_transport_grpc.cpp

```diff
@@ -16,31 +16,36 @@
  */
 #include "ordering/impl/ordering_service_transport_grpc.hpp"
 
+#include "backend/protobuf/from_old_model.hpp"
+#include "builders/protobuf/proposal.hpp"
+#include "datetime/time.hpp"
+
 using namespace iroha::ordering;
-using namespace iroha::protocol;
-using namespace iroha::model;
-using namespace iroha::network;
 
 void OrderingServiceTransportGrpc::subscribe(
-    std::shared_ptr<OrderingServiceNotification> subscriber) {
+    std::shared_ptr<iroha::network::OrderingServiceNotification> subscriber) {
   subscriber_ = subscriber;
 }
 
 grpc::Status OrderingServiceTransportGrpc::onTransaction(
     ::grpc::ServerContext *context,
-    const protocol::Transaction *request,
+    const iroha::protocol::Transaction *request,
     ::google::protobuf::Empty *response) {
   if (subscriber_.expired()) {
     log_->error("No subscriber");
   } else {
-    subscriber_.lock()->onTransaction(*factory_.deserialize(*request));
+    subscriber_.lock()->onTransaction(
+        shared_model::detail::makePolymorphic<shared_model::proto::Transaction>(
```

Comment:
It is necessary to pass rvalue to shared_model proto constructor here, because the object is owned by grpc. So a copy of request should be made.

Reply:
It's const param